model/qti/parser/IncludedElementIdsExtractor.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ */
+
+namespace oat\taoQtiItem\model\qti\parser;
+
+use oat\oatbox\service\ConfigurableService;
+use oat\taoQtiItem\model\qti\Item;
+use tao_helpers_Uri;
+
+class IncludedElementIdsExtractor extends ConfigurableService
+{
+    /** @var string[] */
+    private $ids;
+
+    public function extract(Item $qtiItem): array
+    {
+        $this->ids = [];
+
+        $this->incrementIds($this->getItemBody($qtiItem));
+
+        return array_unique($this->ids);
```

Comment:
does this mean we will drop some versions of the value with different keys ?

Reply:
The ids are unique for media (sharedStimulus) this is the URI id used as reference. The array_unique is to avoid repeat the same id twice.